CMakeLists.txt

```diff
@@ -232,324 +232,326 @@ if(FF_USE_NCCL)
 endif()
 
 # Legion
-include(legion)
-
-# json
-include(json)
-
-# variant
-include(variant)
+if(FF_BUILD_LEGION)
```

Comment:
I think we should rename this flag `PREBUILD_LEGION_FOR_FLEXFLOW`, `BUILD_LEGION_ONLY` or something similar to emphasize that if the flag is set to ON, we do not build FlexFlow, whereas if the flag is set to OFF, we build both Legion and FlexFlow

Reply:
I forgot to mention, we should also declare this variable as an option in the CMakeLists.txt file, and set a default value and description. For this step you can take a look at what we do for the `FF_USE_PYTHON` option, and just copy the code but change the name of the variable